apps/calendar/js/time_observer.js

```diff
@@ -0,0 +1,108 @@
+define(function(require, exports, module) {
+'use strict';
+
+// this module abstracts minute and day changes, it should be used by all the
+// views that needs to listen for time changes (ie. update current time/date).
+// listeners are enabled/disabled when the document visibility changes,
+// reducing battery usage and making sure we always display the proper value.
+// ---
+// XXX: we don't listen to moztimechange here because the app forces a restart
+// when the time changes; even if the app did not restart we would still need
+// to trigger a refresh of most of the views (timezone affects the way we
+// busytimes are rendered in all the views, even the view/edit event views)
+// see: https://bugzilla.mozilla.org/show_bug.cgi?id=1093016#c9
+
+var EventEmitter2 = require('ext/eventemitter2');
+
+exports = module.exports = new EventEmitter2();
+
+exports.on = function() {
+  EventEmitter2.prototype.on.apply(this, arguments);
+  this._start();
+};
+
+exports.once = function() {
+  EventEmitter2.prototype.once.apply(this, arguments);
+  this._start();
```

Comment:
I did not use the `newListener` because `_start` checks if the emitter have any listener (since `_start` is also called when window becomes visible)..